### PR TITLE
Add streamlit example: inference on image folder

### PR DESCRIPTION
**Pull request by AsiaCao**
Add a new streamlit example: inference on image folder

### CODE REVIEW

File: examples/apps/image-folder-support/app.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                class_counts_all[class_name][i] = count
+
+        data = {
+            "image_path": image_paths,
+        }
+        data.update(class_counts_all)
+        return pd.DataFrame(data)
+
+
+def bulk_inference(image_paths: list[str], image_folder_path: str) -> ImageFolderResult:
```
Comment (cat101) [inline examples/apps/image-folder-support/app.py:80]: I think this function would be ideal for the pipeline construct... in fact you are doing a very similar thing to accumulate results on a FrameSet

File: examples/apps/image-folder-support/app.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            labels = [val[1] for val in coverage]
+            sizes = [val[0] for val in coverage]
+            unclassified_pixels = 1 - sum(sizes)
+            sizes.append(unclassified_pixels)
+            labels.append("unclassified")
+            fig = px.pie(
+                values=sizes,
+                names=labels,
+                title="Predicted pixel distribution over all images",
+            )
```
Comment (cat101) [inline examples/apps/image-folder-support/app.py:234]: I we need to add something like Describe() to the pipeline to generate useful and common stats like defect distribution and defect coverage

**Review by cat101 — APPROVED** [review]

File: examples/apps/image-folder-support/app.py (hunk 1 above)
Comment (AsiaCao) [reply examples/apps/image-folder-support/app.py:80]: good point, just add asana task to simplify and refactor this part.
https://app.asana.com/0/0/1204761250364986/f

File: examples/apps/image-folder-support/app.py (hunk 2 above)
Comment (AsiaCao) [reply examples/apps/image-folder-support/app.py:234]: yeah, we can consider supporting plotting and data visualization.
though, i'd wait to see a few more user stories and decide what kind of visualization library are mostly suitable for them. Here I just picked one that I'm familiar with, which may not be the best one.

**Pull request merged**